### PR TITLE
WEB-896 add multi-code-lang page functionality

### PR DESCRIPTION
**Pull request by zbayoff**
### What does this PR do?
Add functionality to swap pages out based on a code language cookie.

The following fronmatter is applicable to each page to be swapped by code lang: 

`type: multi-code-lang` sets the page to use the `multi-code-lang` layout type. 
`code_lang: cpp` set the page code language.
`code_lang_weight: 50` sets the order of the code language that appears in the code lang nav tabs

### Motivation
https://datadoghq.atlassian.net/jira/software/projects/WEB/boards/102?selectedIssue=WEB-896

### Preview
Content reorg starts at this level: https://docs-staging.datadoghq.com/zach/multi-page-code-lang/tracing/setup_overview/

Language-specific pages start at this level: https://docs-staging.datadoghq.com/zach/multi-page-code-lang/tracing/setup_overview/setup/java/?tab=containers

Click a language on this page ^ should go to that language. Each programming language page should have a code lang buttons to select users pref. This will also set a cookie which is compatible with the `programming-lang` shortcode used on main Docs pages and API pages. 

Verify the cookie `code-lang` updates correctly when clicking the different code lang buttons.
Verify the API page code blocks still function as expected.
Verify the pages using the `programming-lang` shortcode function as expected.

### CODE REVIEW

**Review by nsollecito — APPROVED** [review]

**Review by andrewsouthard1 — CHANGES_REQUESTED** [review]
Few comments on clarifying language. In terms of areas that I did not comment on:

https://a.cl.ly/NQu198BN - - The language tiles themselves seem to be much larger than where they are across the documentation, making it jarring.

https://a.cl.ly/4guJGdGP - The Open Standards section overall I think we need to have more a succinct vision on, without some kind of intro to this content it doesn't feel warranted yet to be in it's own area.

File: content/en/tracing/setup_overview/_index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - /tracing/send_traces/
+    - /tracing/setup/
+    - /tracing/environments/
+    - /tracing/setup/environment
+    - /tracing/setup/first_class_dimensions
+    - /tracing/getting_further/first_class_dimensions/
+    - /agent/apm/
+---
+
+Configuring your application to send [traces][1] to Datadog involves two steps:
```
[the remark this answers is not in the document]
Comment (kayayarai) [reply content/en/tracing/setup_overview/_index.md]: Added rewording, fixed language tile widths. The Open Standards improvements are being tracked under DOCS-1409.

File: content/en/tracing/setup_overview/setup/java.md [hunk 2]
```diff
@@ -13,41 +18,97 @@ further_reading:
...
+2. Add the following JVM argument when starting your application in your IDE, Maven or Gradle application script, or `java -jar` command:
 
    ```text
     -javaagent:/path/to/the/dd-java-agent.jar
    ```
 
-4. Add [configuration options](#configuration) for tracing and ensure you are setting environment variables or passing system properties as JVM arguments, particularly for service, environment, logs injection, profiling, and optionally runtime metrics-all the metrics you intend to use. See the examples below. Note that using the in-app quickstart instructions generates these for you.
+3. Add [configuration options](#configuration) for tracing and ensure you are setting environment variables or passing system properties as JVM arguments, particularly for service, environment, logs injection, profiling, and optionally runtime metrics-all the metrics you intend to use. See the examples below. Note that using the in-app quickstart instructions generates these for you.
+
+4. Install and configure the Datadog Agent to receive traces from your now instrumented application:
```
[the remark this answers is not in the document]
Comment (kayayarai) [reply content/en/tracing/setup_overview/setup/java.md]: Done

File: content/en/tracing/setup_overview/_index.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+1. Configuring the Datadog Agent for APM.
+
+2. Adding the Datadog Tracing Library to your code.
+
+These steps are both required for all non-[AWS Lambda][2] environments.  Traces are sent from your application instrumented with a Datadog Tracing Library to the Datadog Agent, and from the Datadog Agent to Datadog.
+
+Depending on your environment, there may be APM-specific configurations required on both the Tracer and Agent to ensure that traces can be properly received, so ensure you have followed instructions for both components.
```
[the remark this answers is not in the document]
Comment (kayayarai) [reply content/en/tracing/setup_overview/_index.md]: Done

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**